arroyo/processing/strategies/collect.py

```diff
@@ -21,7 +32,210 @@ class OffsetRange:
     timestamp: datetime
 
 
+@dataclass(frozen=True)
+class MessageBatch(Generic[TPayload]):
+    """
+    Represents a batch of messages with the range of offsets covered.
+    The range of offsets is represented as a mapping between partitions
+    and offset ranges.
+    The lowest offset in each offset range is inclusize. The highest is
+    esclusive.
+    """
+
+    messages: Sequence[Message[TPayload]]
+    offsets: Mapping[Partition, OffsetRange]
+    duration: float
+
+    def __len__(self) -> int:
+        return len(self.messages)
+
+    def __repr__(self) -> str:
+        return f"<{type(self).__name__}: {len(self)} message{'s' if len(self) != 1 else ''}, open for {self.duration:0.2f} seconds>"
+
+
+class BatchBuilder(Generic[TPayload]):
```

Comment:
You kind of have to read the code of this class to figure out why it exists. As far as I can tell its main purpose is to keep track of the highest offsets per partition of any message. Is it making the assumption in `append` that offsets are strictly increasing? Those assumptions (or any others I've missed) should be stated

Reply:
Behavior changed and comment added